### PR TITLE
adds setting to set non-image-extension image urls

### PR DESCRIPTION
**Pull request by afuggini**
Example RegExp:

`/^https?:\/\/(?:www.)?(?:imgur\.com\/\w{7}|tradingview.com\/\w\/\w{8})\/?$/i`

Parsing function unit-tested here:
https://codepen.io/arielfuggini/pen/YzqVjyg

### CODE REVIEW

File: src/converse-chatview.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ converse.plugins.add('converse-chatview', {
                 'emoji': true,
                 'spoiler': true
             },
+            'image_urls_regex': null
```
Comment (jcbrand) [inline src/converse-chatview.js:70]: When adding a new configuration option, please also document it here:
https://github.com/conversejs/converse.js/blob/master/docs/source/configuration.rst

Options are in alphabetical order.

Also, please provide a test for this. In the test you can pass in a regex for `image_urls_regex` and test that images render.

Your test will be something similar to this:
https://github.com/conversejs/converse.js/blob/823134d7629bad2b82875964169340609c0ce2c9/spec/messages.js#L937

File: src/utils/html.js [hunk 2]
```diff
@@ -75,8 +75,12 @@ function checkFileTypes (types, url) {
 }
 
 u.isAudioURL = url => checkFileTypes(['.ogg', '.mp3', '.m4a'], url);
-u.isImageURL = url => checkFileTypes(['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.tiff', '.svg'], url);
 u.isVideoURL = url => checkFileTypes(['.mp4', '.webm'], url);
+u.isImageURL = url => checkFileTypes(['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.tiff', '.svg'], url);
+u.isWhitelistedImageURL = url => {
+    const regex = _converse.api.settings.get('image_urls_regex');
+    return regex ? regex.test(url) : false;
+}
```
Comment (jcbrand) [inline src/utils/html.js:83]: I don't see any place where `isImageURL` is called without also having to call `isWhitelistedImageURL`, so it seems unnecessary to me to have two methods.
Instead, `isImageURL` can also do the regex test.

If it ever were necessary to not check for the regex, we could disable that by passing in a boolean to `isImageURL`.

**Review by jcbrand — CHANGES_REQUESTED** [review]

**Comment by jcbrand** [comment]
@afuggini There are failing tests due to changes you made in this PR. Please fix.

File: src/utils/html.js (hunk 2 above)
Comment (afuggini) [reply src/utils/html.js:84]: Fixed.

File: src/converse-chatview.js (hunk 1 above)
Comment (afuggini) [reply src/converse-chatview.js]: Done.

**Comment by jcbrand** [comment]
Thanks!

**Pull request merged**